src/cli.rs

```diff
@@ -56,31 +66,115 @@ fn serve_client(rx: &mut impl Read, tx: &mut impl Write, infura_url: &String) {
         let request_id = obj.get("id").unwrap();
         let method = obj.get("method").unwrap();
 
+        let response = match method.as_str().unwrap() {
+            "web3_clientVersion" => format!(
+                r#"{{"jsonrpc":"2.0","id":{},"result":"trin 0.0.1-alpha"}}"#,
+                request_id,
+            )
+            .into_bytes(),
+            _ => {
+                //Re-encode json to proxy to Infura
+                let request = obj.to_string();
+                match proxy_to_url(request, infura_url) {
+                    Ok(result_body) => result_body,
+                    Err(err) => format!(
+                        r#"{{"jsonrpc":"2.0","id":"{}","error":"Infura failure: {}"}}"#,
+                        request_id,
+                        err.to_string(),
+                    )
+                    .into_bytes(),
+                }
+            }
+        };
+        tx.write_all(&response).unwrap();
+    }
+    println!("Clean exit");
+}
+
+fn launch_http_client(pool: ThreadPool, trin_config: TrinConfig) {
+    let uri = format!("127.0.0.1:{}", trin_config.endpoint);
+    let listener = TcpListener::bind(uri).unwrap();
+    for stream in listener.incoming() {
+        match stream {
+            Ok(stream) => {
+                let infura_project_id = trin_config.infura_project_id.clone();
+                pool.execute(move || {
+                    let infura_url =
+                        format!("https://mainnet.infura.io:443/v3/{}", infura_project_id);
```

Comment:
Now that this magic URL string is duplicated, seems like it should be pulled into a new `fn get_infura_url(project_id)`